content/en/docs/components/pipelines/sdk/v2/build-pipeline.md

```diff
@@ -0,0 +1,520 @@
+            pipelines that use this component.
+    
+    This example implements a two-step pipeline.
+
+### Build your pipeline components
+
+        
+Build your pipeline components. This example modifies the initial script to
+extract the contents of a zipped tar file, merge the CSV files that were
+contained in the zipped tar file, and return the merged CSV file.
+
+This example builds a Python function-based component. You can also package
+your component's code as a Docker container image and define the component
+using a ComponentSpec.
+
+In this case, the following modifications were required to the original
+function.
+
+*   The file download logic was removed. The path to the zipped tar file
+    is passed to this function as the `tar_data` argument.
+*   The import statements were moved inside of the function. Python
+    function-based components require standalone Python functions. This
+    means that any required import statements must be defined within the
+    function, and any helper functions must be defined within the function.
+*   The function's arguments are annotated as an [`kfp.dsl.Input`][input]
+    or [`kfp.dsl.Output`][output] artifact. These annotations let Kubeflow
+    Pipelines know to provide the path to the zipped tar file and to
+    create a path where your function stores the merged CSV file.
+*   The function is decorated with the [`kfp.dsl.component`][dsl-component]
+    annotation. You can also use this annotation to define the following:
+    
+    *   The container image that your function runs in.
+    *   Any PyPI packages that this function depends on, that are not already
+        installed on the container image.
+    *   The location to save the component specification to. You can use the
+        component specification to share this component with your colleagues.
+    
+    This annotation converts your function into a factory function that
+    creates pipeline steps. These pipeline steps execute the function you
+    defined as a part of a pipeline's workflow.
+
+Learn more about [building Python function-based components][python-function-components].
+
+The following example shows the updated `merge_csv` function.
+
+[web-download-component]: https://github.com/kubeflow/pipelines/blob/master/components/web/Download/component.yaml
+[python-function-components]: https://www.kubeflow.org/docs/components/pipelines/sdk/v2/python-function-components/
+[input]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+[output]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+[dsl-component]: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_component.py
+
+
+```python
+@component(
+    base_image='python:3.7',
+    packages_to_install=['pandas==1.1.4'],
+    output_component_file='component.yaml' # This is optional. It saves the component spec for future use.
```

Comment:
`base_image='python:3.7',` is also optional since `python:3.7` is the default value.

Reply:
Removed the `base_image='python:3.7'` and the note about the `output_component_file` being optional. The doc on Python function-based components can specify the details.